File: app/src/main/java/com/commit451/gitlab/model/api/ForkedFromProject.java
package com.commit451.gitlab.model.api;

import com.squareup.moshi.Json;

import org.parceler.Parcel;

@Parcel
public class ForkedFromProject {
    @Json(name = "id")
    long id;
    @Json(name = "name")
    String name;
    @Json(name = "name_with_namespace")
    String nameWithNamespace;
    @Json(name = "path")
    String path;
    @Json(name = "path_with_namespace")
    String pathWithNamespace;

    public ForkedFromProject() {}

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getNameWithNamespace() {
        return nameWithNamespace;
    }

    public String getPath() {
        return path;
    }

    public String getPathWithNamespace() {
        return pathWithNamespace;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof ForkedFromProject)) {
            return false;
        }

        ForkedFromProject that = (ForkedFromProject) o;
        return id == that.id;
    }

    @Override
    public int hashCode() {
        return (int) (id ^ (id >>> 32));
    }
}

File: app/src/main/java/com/commit451/gitlab/model/api/ForkedFromProject.kt
package com.commit451.gitlab.model.api

import android.os.Parcelable
import com.squareup.moshi.Json
import kotlinx.android.parcel.Parcelize

@Parcelize
data class ForkedFromProject(
    @Json(name = "id")
    var id: Long = 0,
    @Json(name = "name")
    var name: String? = null,
    @Json(name = "name_with_namespace")
    var nameWithNamespace: String? = null,
    @Json(name = "path")
    var path: String? = null,
    @Json(name = "path_with_namespace")
    var pathWithNamespace: String? = null
) : Parcelable
